### PR TITLE
Feature/scc 443 update title

### PR DESCRIPTION
**Pull request by adrianmilne**
As mentioned - this is all working (seems to be) - but based on your updated YAML - so final solution may change depending on how much Nicks official changes deviate from your experimental ones.

Will sort out an updated Postman too

### CODE REVIEW

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementEventService.java [hunk 1]
```diff
@@ -156,6 +158,76 @@ private CreateUpdateRfx createUpdateRfxRequest(final ProcurementProject project,
...
+      final String eventName, final String principal) {
+
+    Assert.hasLength(eventName, "New event name must be supplied");
+
+    // EventId will be in the format "ocds-b5fd17-1"
+    String ocdsAuthorityName;
+    String ocidPrefix;
+    Integer eventIdKey;
+
+    // Validate eventId
```
Comment (adrianmilne) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementEventService.java:178]: Wasn't thrilled with this bit (on extracting the individual items from the concatenated id)- did experiment with using an @Embeddable to make all 3 of these fields into an object that could be treated as a single item for converting to/from the concatenated ID - but given one was the primary @Id - didn't seem to like it - and then thought it was maybe overkill. Happily take any suggestions to improve

Comment (tommyb82) [reply src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementEventService.java:178]: Only that perhaps matching a regex pattern then extracting the capture groups *might* be more reliable / concise.. or just more of a headache! 😃

Comment (adrianmilne) [reply src/main/java/uk/gov/crowncommercial/dts/scale/cat/service/ProcurementEventService.java:178]: i tend to equate regex with headaches - but I will take a look - certainly should be more concise 😃

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/controller/EventsController.java [hunk 2]
```diff
@@ -27,9 +30,24 @@ public EventSummary createProcurementEvent(@PathVariable("procID") Integer procI
       @RequestBody Tender tender, JwtAuthenticationToken authentication) {
 
     var principal = authentication.getTokenAttributes().get("sub").toString();
-    log.info("createProcuremenProject invoked on behalf of principal: {}", principal);
+    log.info("createProcuremenEvent invoked on behalf of principal: {}", principal);
 
     return procurementEventService.createFromTender(procId, tender, principal);
   }
 
+  @PostMapping(value = "/tenders/projects/{procID}/events/{eventID}/name",
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/controller/EventsController.java:38]: Aha! Tis a `PUT` according to API spec I believe 🤓

File: src/main/java/uk/gov/crowncommercial/dts/scale/cat/controller/ProjectsController.java [hunk 3]
```diff
@@ -31,4 +35,17 @@ public DraftProcurementProject createProcurementProject(
     return procurementProjectService.createFromAgreementDetails(agreementDetails, principal);
   }
 
+  @PostMapping(value = "/tenders/projects/{procID}/name",
```
Comment (tommyb82) [inline src/main/java/uk/gov/crowncommercial/dts/scale/cat/controller/ProjectsController.java:38]: As above - a `PUT` I think..

**Review by tommyb82 — CHANGES_REQUESTED** [review]
Couldn't find much.. looking grand as always! 👍

**Comment by tommyb82** [comment]
With the `@PutMapping`, also need to update the [security config](https://github.com/Crown-Commercial-Service/ccs-scale-cat-service/blob/feature/SCC-443_update_title/src/main/java/uk/gov/crowncommercial/dts/scale/cat/config/OAuth2Config.java#L29) - could just specify all methods i.e.
`.antMatchers("/tenders/**").hasAnyAuthority("CAT_USER", "CAT_ADMINISTRATOR")`

**Comment by adrianmilne** [comment]
> With the `@PutMapping`, also need to update the [security config](https://github.com/Crown-Commercial-Service/ccs-scale-cat-service/blob/feature/SCC-443_update_title/src/main/java/uk/gov/crowncommercial/dts/scale/cat/config/OAuth2Config.java#L29) - could just specify all methods i.e.
> `.antMatchers("/tenders/**").hasAnyAuthority("CAT_USER", "CAT_ADMINISTRATOR")`

Yep - got that one thanks (wasn't sure whether to distinguish between HTTP verb - but will just do a catch all like you suggest) 👍

**Review by tommyb82 — APPROVED** [review]
👍 Excellent work - have updated sbx2, dev and int DBs so builds should now succeed

**Pull request merged**